stl/inc/ranges

```diff
@@ -7656,6 +7662,401 @@ namespace ranges {
+            }
+        }();
+
+        static constexpr bool _Enable_const_begin_end =
+            (range<const _Inner_view> && regular_invocable<const _Func&, range_reference_t<const _ViewTypes>...>);
+
+    public:
+        zip_transform_view() noexcept(
+            is_nothrow_default_constructible_v<_Movable_box<_Func>>&& is_nothrow_default_constructible_v<_Inner_view>) =
+            default;
+
+        constexpr explicit zip_transform_view(_Func _Function_, _ViewTypes... _Views) noexcept(
+            is_nothrow_move_constructible_v<_Movable_box<_Func>>&&
+                is_nothrow_constructible_v<_Inner_view, remove_reference_t<_ViewTypes>&&...>) // strengthened
+            : _Function(in_place, _STD move(_Function_)), _Zip(_STD move(_Views)...) {}
+
+        constexpr auto begin() noexcept(noexcept(_Iterator<false>{*this, _Zip.begin()})) // strengthened
+        {
+            return _Iterator<false>{*this, _Zip.begin()};
+        }
+
+        constexpr auto begin() const noexcept(noexcept(_Iterator<true>{*this, _Zip.begin()})) // strengthened
+            requires _Enable_const_begin_end
+        {
+            return _Iterator<true>{*this, _Zip.begin()};
+        }
+
+        constexpr auto end() noexcept(_Is_end_noexcept<false>) // strengthened
+        {
+            if constexpr (common_range<_Inner_view>) {
+                return _Iterator<false>{*this, _Zip.end()};
+            } else {
+                return _Sentinel<false>{_Zip.end()};
+            }
+        }
+
+        constexpr auto end() const noexcept(_Is_end_noexcept<true>) // strengthened
+            requires _Enable_const_begin_end
+        {
+            if constexpr (common_range<const _Inner_view>) {
+                return _Iterator<true>{*this, _Zip.end()};
+            } else {
+                return _Sentinel<true>{_Zip.end()};
+            }
+        }
+
+        constexpr auto size() noexcept(noexcept(_Zip.size())) // strengthened
+            requires sized_range<_Inner_view>
+        {
+            return _Zip.size();
+        }
+
+        constexpr auto size() const noexcept(noexcept(_Zip.size())) // strengthened
+            requires sized_range<const _Inner_view>
+        {
+            return _Zip.size();
+        }
+    };
+
+    template <class _Func, class... _Ranges>
+    zip_transform_view(_Func, _Ranges&&...) -> zip_transform_view<_Func, views::all_t<_Ranges>...>;
+
+    namespace views {
+        struct _Zip_transform_fn {
+        private:
+            template <class _Func, class... _Types>
+            static constexpr bool _Is_invocation_noexcept = []() noexcept {
+                if constexpr (sizeof...(_Types) == 0) {
+                    // equivalent to:
+                    // using _Decayed_func = decay_t<_Func>;
+                    //
+                    // return noexcept(((void) _STD declval<_Func&>(),
+                    //     auto(empty<decay_t<invoke_result_t<_Decayed_func&>>>)));

```

Comment:
@CaseyCarter and I think that it would be simpler to drop this comment entirely. In the current comment, `(void) _STD declval<_Func&>()` is extremely strange (this is always `noexcept`, so there's no reason to mention it here). We saw that it was trying to match `((void)F,` in the Standardese, but readers of this implementation should already know to be looking at the Standardese side-by-side.